server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -372,25 +385,25 @@ public void testAckListenerReceivesNacksIfPublicationTimesOut() {
     }
 
     public void testAckListenerReceivesNacksIfLeaderStandsDown() {
-        // TODO: needs support for handling disconnects
-//        final Cluster cluster = new Cluster(3);
-//        cluster.runRandomly();
-//        cluster.stabilise();
-//        final ClusterNode leader = cluster.getAnyLeader();
-//        final ClusterNode follower0 = cluster.getAnyNodeExcept(leader);
-//        final ClusterNode follower1 = cluster.getAnyNodeExcept(leader, follower0);
-//
-//        leader.partition();
-//        follower0.coordinator.handleDisconnectedNode(leader.localNode);
-//        follower1.coordinator.handleDisconnectedNode(leader.localNode);
-//        cluster.runUntil(cluster.getCurrentTimeMillis() + cluster.DEFAULT_ELECTION_TIME);
-//        AckCollector ackCollector = leader.submitRandomValue();
-//        cluster.runUntil(cluster.currentTimeMillis + Cluster.DEFAULT_DELAY_VARIABILITY);
-//        leader.connectionStatus = ConnectionStatus.CONNECTED;
-//        cluster.stabilise(cluster.DEFAULT_STABILISATION_TIME, 0L);
-//        assertTrue("expected nack from " + leader, ackCollector.hasAckedUnsuccessfully(leader));
-//        assertTrue("expected nack from " + follower0, ackCollector.hasAckedUnsuccessfully(follower0));
-//        assertTrue("expected nack from " + follower1, ackCollector.hasAckedUnsuccessfully(follower1));
+        final Cluster cluster = new Cluster(3);
+        cluster.runRandomly();
+        cluster.stabilise();
+        final ClusterNode leader = cluster.getAnyLeader();
+        final ClusterNode follower0 = cluster.getAnyNodeExcept(leader);
+        final ClusterNode follower1 = cluster.getAnyNodeExcept(leader, follower0);
+
+        leader.blackhole();
+        follower0.onDisconnectEventFrom(leader);
+        follower1.onDisconnectEventFrom(leader);
+        cluster.runFor(DEFAULT_DELAY_VARIABILITY // disconnect is scheduled
+            + DEFAULT_ELECTION_DELAY, "elect new leader");
```

Comment:
we want the 2 followers to complete an election and elect a leader among themselves before "healing" the leader (and prevent the leader from preventing that election), so that the publishing of the subsequent value will fail. I've added a comment.